### PR TITLE
Update to handle aborted with success

### PR DESCRIPTION
**Pull request by hisaac**
### Checklist

- [x] I've read and followed the [Contribution Guidelines](https://github.com/bitrise-steplib/.github/blob/main/CONTRIBUTING.md)
- [x] `step.yml` and `README.md` is updated with the changes (if needed)

### Version

Requires a *MINOR* [version update](https://semver.org/)

### Context

Related to a recent change to the build-router-start Step (https://github.com/bitrise-steplib/bitrise-step-build-router-start/pull/35), we need to update this Step to handle the "Aborted with success" status.

### Changes

- Updates bitrise-step-build-router-start module to 0.14.5
- Updates `main.go` to check for failed and aborted statuses using new convenience methods
- Updates other uses of `build.status` to use new convenience methods

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -46,23 +46,23 @@ func main() {
...
 			log.Donef("- %s successful %s)", build.TriggeredWorkflow, buildURL)
-		case 2:
+		} else if build.IsFailed() {
 			log.Errorf("- %s failed", build.TriggeredWorkflow)
 			failReason = "failed"
-		case 3:
+		} else if build.IsAborted() {
 			log.Warnf("- %s aborted", build.TriggeredWorkflow)
 			failReason = "aborted"
-		case 4:
```
Comment (hisaac) [inline main.go:60]: @gaborszakacs This was previously set to recognize a status of `4` as "Cancelled". Did this change to mean "Aborted With Success"?

Comment (gaborszakacs) [reply main.go:60]: Status 4 has always been "Aborted With Success", it's a bit confusing name (and concept), we might use "Cancelled" on the UI for this, which is similarly confusing IMO..

Comment (hisaac) [reply main.go:60]: ok, I guess for now I'll just leave the log message as-is then

File: main.go [hunk 2]
```diff
@@ -46,23 +46,23 @@ func main() {
...
+		} else if build.IsAbortedWithSuccess() {
 			log.Infof("- %s cancelled", build.TriggeredWorkflow)
 			failReason = "cancelled"
```
Comment (gaborszakacs) [inline main.go]: We don't need to set this `failReason` here as it won't go into the abort branch where we would use it.

Comment (hisaac) [reply main.go]: Ah good catch. Updated.

**Review by gaborszakacs — APPROVED** [review]

**Pull request merged**